tests/test_openshift.py

```diff
@@ -0,0 +1,455 @@
+
+
+@pytest.fixture(scope=FIXTURES_SCOPES)
+def gen_image(provider):
+    if MOCKED:
+        return Image(provider, 'some.test.image', 'sha256:{}'
+                     .format(fauxfactory.gen_alphanumeric(64)))
+    return choice(provider.list_docker_image())
+
+
+@pytest.fixture(scope=FIXTURES_SCOPES)
+def gen_pod(provider):
+    if MOCKED:
+        return Pod(provider, 'some-test-pod', 'default')
+    return choice(provider.list_container_group())
+
+
+@pytest.fixture(scope=FIXTURES_SCOPES)
+def gen_service(provider):
+    if MOCKED:
+        return Service(provider, 'some-test-service', 'default')
+    return choice(provider.list_service())
+
+
+@pytest.fixture(scope=FIXTURES_SCOPES)
+def gen_node(provider):
+    if MOCKED:
+        return Node(provider, 'openshift-node.test.com')
+    return choice(provider.list_node())
+
+
+@pytest.fixture(scope=FIXTURES_SCOPES)
+def gen_replicator(provider):
+    if MOCKED:
+        return Replicator(provider, 'some-test-replicator', 'default')
+    return choice(provider.list_replication_controller())
+
+
+@pytest.fixture(scope=FIXTURES_SCOPES)
+def gen_route(provider):
+    if MOCKED:
+        return Route(provider, 'some.test.route.com', 'default')
+    return choice(provider.list_route())
+
+
+@pytest.fixture(scope=FIXTURES_SCOPES)
+def gen_template(provider):
+    if MOCKED:
+        return Template(provider, 'some-test-template', 'default')
+    return choice(provider.list_template())
+
+
+@pytest.fixture(scope=FIXTURES_SCOPES)
+def gen_image_registry(provider):
+    return ImageRegistry(provider, 'openshift-hello-openshift',
+                        'docker.io/openshift/hello-openshift', 'default')
+
+
+@pytest.fixture(scope=FIXTURES_SCOPES)
+def gen_volume(provider):
+    if MOCKED:
+        return Volume(provider, 'my-test-persistent-volume')
+    return choice(provider.list_volume())
+
+
+@pytest.fixture(scope=FIXTURES_SCOPES)
+def gen_dc(provider):
+    return DeploymentConfig(provider, fauxfactory.gen_alpha().lower(),
+                            'default', 'openshift/hello-openshift', 1)
+
+
+@pytest.fixture(scope=FIXTURES_SCOPES)
+def label():
+    return (fauxfactory.gen_alpha().lower(), fauxfactory.gen_alpha().lower())
+
+
+def base__test_label_create(resource, label_key, label_value):
+    if MOCKED:
+        resource.provider.api.patch.return_value = \
+            resource.provider.o_api.patch.return_value = [201, {}]
+        resource.provider.api.get.return_value = \
+            resource.provider.o_api.get.return_value = [200, {
+                'metadata': {'labels': {label_key: label_value}}}]
+    res = resource.set_label(label_key, label_value)
+    assert res[0] in (200, 201)
+    assert wait_for(lambda: label_key in resource.list_labels(),
+                    message="Waiting for label {} of {} {} to be exist..."
+                    .format(label_key, resource.__class__.__name__, resource.name),
+                    delay=5, timeout='1M')
```

Comment:
make a newline before lambda, that will free up some space on the line, and use `type(resource)` instead of `resource.__class__`. That may then fit the `.format` on the previous line, making it look better.